packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -985,6 +990,24 @@ def response_headers_and_deserialization(
                 retval.append(f"    deserialized = None")
         return retval
 
+    @staticmethod
+    def _need_deserialize(t: BaseType) -> bool:
```

Comment:
I think we should pass all of these types to the `_deserialize` function, and then the `_deserialize` function can decide whether to deserialize or not

Reply:
done~